src/tensors/linalg.jl

```diff
@@ -45,106 +45,104 @@ function one!(t::AbstractTensorMap)
-    InnerProductStyle(S) === EuclideanProduct() || throw_invalid_innerproduct(:unitary)
-    return isomorphism(P)
-end
-function unitary(A::Type{<:DenseMatrix}, P::TensorMapSpace{S}) where {S}
-    InnerProductStyle(S) === EuclideanProduct() || throw_invalid_innerproduct(:unitary)
-    return isomorphism(A, P)
-end
-function unitary(A::Type{<:DenseMatrix}, cod::TensorSpace{S}, dom::TensorSpace{S}) where {S}
+function unitary(::Type{A}, V::TensorMapSpace{S,N₁,N₂}) where {A<:MatOrNumber,S,N₁,N₂}
     InnerProductStyle(S) === EuclideanProduct() || throw_invalid_innerproduct(:unitary)
-    return isomorphism(A, cod, dom)
+    return isomorphism(A, V)
 end
 
 """
-    isometry([A::Type{<:DenseMatrix} = Matrix{Float64},] cod::VectorSpace, dom::VectorSpace)
-    -> TensorMap
-
-Return a `t::TensorMap` that implements a specific isometry that embeds the domain `dom`
-into the codomain `cod`, and which requires that `spacetype(dom)` (`== spacetype(cod)`) has
-an Euclidean inner product. An isometry `t` is such that its adjoint `t'` is the left
-inverse of `t`, i.e. `t'*t = id(dom)`, while `t*t'` is some idempotent endomorphism of
-`cod`, i.e. it squares to itself. When `dom` and `cod` do not allow for such an isometric
-inclusion, an error will be thrown.
+    isometry([T::Type=Float64,] codomain::TensorSpace, domain::TensorSpace) -> TensorMap
+    isometry([T::Type=Float64,] codomain ← domain) -> TensorMap
+    isometry([T::Type=Float64,] domain → codomain) -> TensorMap
+
+Construct a specific isometry between the codomain and the domain, i.e. return a
+`t::TensorMap` where either `scalartype(t) = T` if `T` is a `Number` type or
+`storagetype(t) = T` if `T` is a `DenseMatrix` type. The isometry `t` then satisfies
+`t' * t = id(domain)` and `(t * t')^2 = t * t'`. If the spaces do not allow for such an 
+isometric inclusion, an error will be thrown.
+
+See also [`isomorphism`](@ref) and [`unitary`](@ref).
 """
-isometry(cod::TensorSpace, dom::TensorSpace) = isometry(Matrix{Float64}, cod, dom)
-isometry(P::TensorMapSpace) = isometry(codomain(P), domain(P))
-isometry(A::Type{<:DenseMatrix}, P::TensorMapSpace) = isometry(A, codomain(P), domain(P))
-function isometry(A::Type{<:DenseMatrix}, cod::TensorSpace, dom::TensorSpace)
-    return isometry(A, convert(ProductSpace, cod), convert(ProductSpace, dom))
-end
-function isometry(::Type{A},
-                  cod::ProductSpace{S},
-                  dom::ProductSpace{S}) where {A<:DenseMatrix,S<:ElementarySpace}
+function isometry(::Type{A}, V::TensorMapSpace{S,N₁,N₂}) where {A<:MatOrNumber,S,N₁,N₂}
     InnerProductStyle(S) === EuclideanProduct() || throw_invalid_innerproduct(:isometry)
-    dom ≾ cod ||
-        throw(SpaceMismatch("codomain $cod and domain $dom do not allow for an isometric mapping"))
-    t = TensorMap(s -> A(undef, s), cod, dom)
-    for (c, b) in blocks(t)
+    domain(V) ≾ codomain(V) ||
+        throw(SpaceMismatch("codomain and domain are not isometric: $V"))
```

Comment:
"isometric" means equivalent as metric spaces. For inner product spaces, this basically means that they are unitarily equivalent. Hence, I prefer the older terminology.